addons/promote/common/src/main/data/promote/rules/npm-version-pattern.groovy

```diff
@@ -0,0 +1,45 @@
+package org.commonjava.indy.promote.rules
+
+import org.apache.commons.lang.StringUtils
+import org.commonjava.indy.pkg.PackageTypeConstants;
+import org.commonjava.indy.promote.validate.model.ValidationRequest
+import org.commonjava.indy.promote.validate.model.ValidationRule
+import org.slf4j.LoggerFactory
+
+class NPMVersionPattern implements ValidationRule {
+
+    String validate(ValidationRequest request) throws Exception {
+        def versionPattern = request.getValidationParameter("versionPattern")
+        def errors = Collections.synchronizedList(new ArrayList())
+        def logger = LoggerFactory.getLogger(getClass())
+        if (request.getTarget().getPackageType() == PackageTypeConstants.PKG_TYPE_NPM && request.getSource().getPackageType() == PackageTypeConstants.PKG_TYPE_NPM) {
+            logger.info("Start to do NPMVersionPattern validation check for request paths: {}", request.getSourcePaths())
+            if (versionPattern == null) {
+                logger.info("No 'versionPattern' parameter specified in rule-set: {}, will only check redhat scoped rule.", request.getRuleSet().getName())
+            }
+            def tools = request.getTools()
+            tools.paralleledEach(request.getSourcePaths(), { it ->
+                def pkg = tools.getNPMPackagePath(it)
+                if (pkg.isPresent()) {
+                    def pkgPath = pkg.get()
+                    def isRedHatScoped = pkgPath.scoped && pkgPath.scopedName == "@redhat"
```

Comment:
Since ythis is going into the public Indy codebase I would prefer to have the scope configurable instead of hardcoded "@redhat".

Reply:
Ok, sounds good.